### PR TITLE
PLIP 2780: textindexer

### PR DESCRIPTION
**Pull request by jensens**
Integration of `collective.textindexer` -> `plone.app.dexterity.textindexer`.
https://github.com/plone/Products.CMFPlone/issues/2780

### CODE REVIEW

**Review by MrTango — CHANGES_REQUESTED** [review]
docs need to be updated, broken links

**Comment by jensens** [comment]
@jenkins-plone-org please run jobs

File: docs/advanced/catalog-indexing-strategies.rst [hunk 1]
```diff
@@ -12,17 +12,17 @@ Making content searchable
...
+Add ``plone.app.dexterity`` to your buildout and you will gain the new Dexterity behavior `plone.textindexer` that will allow you to easily add fields to SearchableText. Once you turn on this behavior, you will then need to specify fields for addition to SearchableText.
 
 .. note::
 
     Note that if you turn on the ``Dynamic SearchableText indexer behavior`` for a content type, then you must specify all fields that need SearchableText indexing. Dublin core fields like Title and Description are no longer automatically handled.
 
 Once you have turned on the indexer behavior, edit the XML field model to add ``indexer:searchable="true"`` to the ``field`` tag for each field you wish to add to the SearchableText index.
 
-See the `collective.dexteritytextindexer  <https://github.com/collective/collective.dexteritytextindexer>`_ package documentation for details and for information on how to use it via Python schema.
+See the `plone.app.dexterity.textindexer  <https://github.com/collective/plone.app.dexterity.textindexer>`_ package documentation for details and for information on how to use it via Python schema.
```
Comment (MrTango) [inline docs/advanced/catalog-indexing-strategies.rst:25]:
```suggestion
See the `/backend/indexing  <https://6-dev-docs.plone.org/backend/indexing>`_ documentation for details and for information on how to use it via Python schema.
```

File: docs/advanced/catalog-indexing-strategies.rst [hunk 2]
```diff
@@ -12,17 +12,17 @@ Making content searchable
 
 Plone's main index is called *SearchableText*. This is the index which is searched when you use the main portal search. Fields in your custom content types are not necessarily added to SearchableText. Fields added via Dublin-core behaviors are automatically part of SearchableText; others are not.
 
-So, you may need to explicitly add fields to SearchableText if you wish their information to be findable via the main search. There are all sorts of highly customizable ways to do this, but the easiest is to use the `collective.dexteritytextindexer  <https://github.com/collective/collective.dexteritytextindexer>`_ add-on package.
+So, you may need to explicitly add fields to SearchableText if you wish their information to be findable via the main search. There are all sorts of highly customizable ways to do this, but the easiest is to use the `plone.app.dexterity.textindexer  <https://github.com/collective/plone.app.dexterity.textindexer>`_ add-on package.
```
Comment (MrTango) [inline docs/advanced/catalog-indexing-strategies.rst:15]:
```suggestion
So, you may need to explicitly add fields to SearchableText if you wish their information to be findable via the main search. There are all sorts of highly customizable ways to do this, but the easiest is to use the `plone.app.dexterity.textindexer  <https://github.com/plone/plone.app.dexterity>`_ core package.
```

**Review by MrTango — APPROVED** [review]
Looks good, i suggested some link fixes to the new documentation, where the updated docs from collective.dexteritytextindexer live now, than it should be good to go.

**Comment by MrTango** [comment]
@pbauer you where on the failing tests, right?

**Comment by pbauer** [comment]
tests are fixed and docs are updated.

**Pull request merged**